opentelemetry-user-events-trace/src/trace/mod.rs

```diff
@@ -0,0 +1,51 @@
+use exporter::UserEventsSpanExporter;
+use opentelemetry::otel_warn;
+use opentelemetry_sdk::trace::TracerProviderBuilder;
+use reentrant_spanprocessor::ReentrantSpanProcessor;
+
+mod exporter;
+mod reentrant_spanprocessor;
+
+/// Extension trait for adding a user event span exporter to the tracer provider builder.
+pub trait UserEventsTracerProviderBuilderExt {
+    /// Adds a user event span exporter to the tracer provider builder with the given provider name.
+    ///
+    /// The provider name must:
+    /// - Be less than 234 characters.
+    /// - Contain only ASCII letters, digits, and the underscore (`'_'`) character.
+    /// - Be short, human-readable, and unique enough to avoid conflicts with other provider names.
+    /// - Typically include a company name and a component name, e.g., "MyCompany_MyComponent".
+    ///
+    /// If an invalid provider name is provided, this method will not add the exporter to the builder.
+    ///
+    /// Tracepoint names are generated by combining the provider name, event
+    /// level and keyword (currently hardcoded to `1`) in the following format:
+    /// `ProviderName + '_' + 'L' + EventLevel + 'K' + EventKeyword`
+    ///
+    /// For example, if "myprovider" is the provider name, the following tracepoint names are created:
```

Comment:
this needs to be adjusted for Span as there is no "level" concept.

Reply:
Done.